cmake/post_build_helper/compile_shaders.cmake

```diff
@@ -0,0 +1,62 @@
+function(get_shaders shaders glslDirectory)
+    file(GLOB_RECURSE glslSourceFiles RELATIVE ${glslDirectory}
+        "${glslDirectory}/**.vert"
+        "${glslDirectory}/**.tesc"
+        "${glslDirectory}/**.tese"
+        "${glslDirectory}/**.geom"
+        "${glslDirectory}/**.frag"
+        "${glslDirectory}/**.comp"
+        "${glslDirectory}/**.mesh"
+        "${glslDirectory}/**.task"
+        "${glslDirectory}/**.rgen"
+        "${glslDirectory}/**.rint"
+        "${glslDirectory}/**.rahit"
+        "${glslDirectory}/**.rchit"
+        "${glslDirectory}/**.*.glsl")
+    set(${shaders} ${glslSourceFiles} PARENT_SCOPE)
+endfunction(get_shaders)
+
+function(make_shader_target shaderTarget target glslDirectory spvDirectory)
+    find_program(glslangValidator glslangValidator
+        HINTS "$ENV{VULKAN_SDK}/Bin" "$ENV{VULKAN_SDK}/x86_64/bin"
+        DOC "The glslangValidator executable.")
+
+    get_shaders(glslSourceFiles ${glslDirectory})
+
+    foreach(glslShader ${glslSourceFiles})
+        set(spvShaderFile "${spvDirectory}/${glslShader}.spv")
+        set(glslShaderAbsolute "${glslDirectory}/${glslShader}")
+        get_filename_component(glslShaderFileName ${glslShader} NAME)
+        get_filename_component(spvShaderDirectory ${spvShaderFile} DIRECTORY)
+        add_custom_command(
+            OUTPUT ${spvShaderFile}
+            COMMAND ${CMAKE_COMMAND} -E make_directory ${spvShaderDirectory}
+            COMMAND ${glslangValidator}
+                --target-env vulkan1.2 -V ${glslShaderAbsolute} -o ${spvShaderFile}
+            DEPENDS ${glslShaderAbsolute}
+            COMMENT "Compiling GLSL shader ${glslShaderFileName} to SPIR-V shader ${spvShaderFile}")
+        list(APPEND spvShaders ${spvShaderFile})
+        message(STATUS "Added GLSL shader ${glslShaderFileName} to dependencies of target ${target}")
+    endforeach(glslShader)
+    add_custom_target(${shaderTarget}
+            DEPENDS ${spvShaders})
+    add_dependencies(${target}
+            ${shaderTarget})
+endfunction()
+
+function(compile_shaders targetName glslDirectory spvDirectory)
+    find_program(glslangValidator glslangValidator
+        HINTS "$ENV{VULKAN_SDK}/Bin" "$ENV{VULKAN_SDK}/x86_64/bin"
+        DOC "The glslangValidator executable.")
+
+    get_shaders(glslSourceFiles ${glslDirectory})
+
+    foreach(glslShader ${glslSourceFiles})
+        set(spvShaderFile "${spvDirectory}/${glslShader}.spv")
+        get_filename_component(glslShaderFileName ${glslShader} NAME)
+        get_filename_component(spvShaderDirectory ${spvShaderFile} DIRECTORY)
+        execute_process(COMMAND ${glslangValidator}
+            --target-env vulkan1.2 -V "${glslDirectory}/${glslShader}" -o ${spvShaderFile})
+        message(STATUS "Compiled GLSL shader ${glslShaderFileName} to SPIR-V shader ${spvShaderFile}")
+    endforeach(glslShader)
+endfunction()
```

Comment:
Is it best practice for CMake files, too, to add newlines at the end of files?

Reply:
Yes, every file should end with a newline.
fixed in https://github.com/cg-tuwien/Gears-Vk/pull/109/commits/5b7f3c20235885ba1c8a5c3be986e10cfee16a86